packages/framework/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -253,4 +254,85 @@ describe("AzureClient", () => {
         const obj = await map1.get("new-pair-id").get();
         assert.ok(obj, "container added dynamic objects incorrectly");
     });
+
+    /**
+     * Scenario: test if connect()/disconnect() can manage a container's connection state
+     *
+     * Expected behavior: the container should be able to connect and disconnect. The connectionState
+     * value should be updated accordingly. No errors should be thrown.
+     */
+    it("can disconnect() and connect() a container", async () => {
+        const container = (await client.createContainer(schema)).container;
+        await container.attach();
+        await new Promise<void>((resolve) => {
+            container.once("connected", () => {
+                resolve();
+            });
+        });
+
+        container.disconnect?.();
+        assert.strictEqual(container.connectionState, ConnectionState.Disconnected, "container can't disconnect");
+
+        container.connect?.();
+        await new Promise<void>((resolve) => {
+            container.once("connected", () => {
+                resolve();
+            });
+        });
+        assert.strictEqual(
+            container.connectionState, ConnectionState.Connected,
+            "container can't connect after calling disconnect()",
+        );
+    });
+
+    /**
+     * Scenario: test that ops processing stops when disconnect() is called, and it resumes
+     * when connect() is called.
+     *
+     * Expected behavior: We should not be able to get the updated map value after disconnect() is called.
+     * Once connect() is called and the container is reconnected we should be able to fetch the updated value.
+     */
+    it("can disconnect() to prevent processing ops and connect() to catch up", async () => {
+        const containerCreate = (await client.createContainer(schema)).container;
+        const containerId = await containerCreate.attach();
+        await new Promise<void>((resolve) => {
+            containerCreate.once("connected", () => {
+                resolve();
+            });
+        });
+
+        const initialObjectsCreate = containerCreate.initialObjects;
+        const map1Create = initialObjectsCreate.map1 as SharedMap;
+        map1Create.set("key", "value");
+        let valueCreate = await map1Create.get("key");
+
+        const containerGet = (await client.getContainer(containerId, schema)).container;
+        const map1Get = containerGet.initialObjects.map1 as SharedMap;
+        if (containerGet.connectionState !== ConnectionState.Connected) {
```

Comment:
Can you assert about this, or is it indeterminate from run to run...?

Reply:
Yes, I believe it is indeterminate. Sometimes the container can connect almost immediately, and other times we may need to wait a small amount of time (usually under 1 second).

Reply:
Interesting, ok.

btw, I think you can actually skip the if check, because Container has fancy logic when you add a listener to the `connected` event it will fire the event if it's connected.  But it's not wrong as-is, JFYI :)

Reply:
I would never rely on that special-case, it's an antipattern for event handling IMO.